File: go/errorTest2.go
package main
import "fmt"

func main(){
	test()
}

type AppleError struct{
	code string
	message string
}

func (err AppleError) Error()(string){
	return fmt.Sprintf(`code: %s, message: %s`, err.code, err.message)
}

func test(){
	var fa = func()(res string, err error){
		err = AppleError{"2020", "apple"}
		return
	}
	res, err := fa()
	if err != nil{
		panic(err)
	}
	println(res)
}

File: python/errorTest2.py

class MyError(Exception):
	def __init__(self, code, message):
		self.code=code
		self.message=message

	def __str__(self):
		return str("%s %s" %(self.code, self.message))

def test():
	raise MyError("S800", "testing error")

def test2():
	try:
		raise MyError("S800", "testing error")
	except MyError as e:
		print("error", e)

test2()